### PR TITLE
add rename command to maptool

### PR DESCRIPTION
**Pull request by jfmherokiller**
This commit adds a rename subcommand to the maptool command in order to allow the the user to rename the maps without going through the gui. 

This was done because the included library [zLib](https://github.com/zDevelopers/zLib) sometimes has issues pulling up the sign based gui.

### CODE REVIEW

**Comment by Adam759** [comment]
I wish the developers of this plugin were still around, this would be great functionality that would really complete this plugin. I may just recompile the jar with this change.

**Comment by AmauryCarrade** [comment]
Hi! We're back.

I approve this functionality. I improved zLib's sign based GUI compatibility, but an alternative for thoses liking commands is always great.

The only problem with this PR is that _map names can include spaces_. A fix for this would be to allows to use quotation marks around the names, and to consider them without spaces it there isn't quotes. Like:

```
/maptool rename Map Map2
/maptool rename Map "Another name for the map"
/maptool rename "Another name for the map" "This one is better"
/maptool rename "This one is better" Better
```

Are you still interested in this or not?

File: src/main/java/fr/moribus/imageonmap/ImageOnMap.java [hunk 1]
```diff
@@ -18,13 +18,7 @@
...
 package fr.moribus.imageonmap;
 
-import fr.moribus.imageonmap.commands.maptool.DeleteCommand;
-import fr.moribus.imageonmap.commands.maptool.ExploreCommand;
-import fr.moribus.imageonmap.commands.maptool.GetCommand;
-import fr.moribus.imageonmap.commands.maptool.GetRemainingCommand;
-import fr.moribus.imageonmap.commands.maptool.ListCommand;
-import fr.moribus.imageonmap.commands.maptool.MigrateCommand;
-import fr.moribus.imageonmap.commands.maptool.NewCommand;
+import fr.moribus.imageonmap.commands.maptool.*;
```
Comment (AmauryCarrade) [inline src/main/java/fr/moribus/imageonmap/ImageOnMap.java]: Please avoir star imports :)

File: src/main/java/fr/moribus/imageonmap/commands/maptool/RenameCommand.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+public class RenameCommand extends IoMCommand {
+    @Override
+    protected void run() throws CommandException
+    {
+        if(args.length == 2)
+        {
+            ImageMap map = getMapFromArgs();
+            map.rename(args[1]);
+        } else {
+            info("Not Enough Or To Many Arguments");
```
Comment (AmauryCarrade) [inline src/main/java/fr/moribus/imageonmap/commands/maptool/RenameCommand.java]: Please use the translations system: `info(I.t("Not enough or too many arguments"));`.

This comment is likely to become irrelevant for _this_ string if you fix the PR to support names with spaces, but it's a general comment on all strings displayed to a user :) .

File: src/main/java/fr/moribus/imageonmap/commands/maptool/RenameCommand.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        }
+    }
+    @Override
+    protected List<String> complete() throws CommandException
+    {
+        if(args.length == 1)
+            return getMatchingMapNames(playerSender(), args[0]);
+        return null;
+    }
```
Comment (AmauryCarrade) [inline src/main/java/fr/moribus/imageonmap/commands/maptool/RenameCommand.java:48]: Please add the permission check:

```java
@Override
public boolean canExecute(CommandSender sender)
{
    return Permissions.RENAME.grantedTo(sender);
}
```

**Review by AmauryCarrade — CHANGES_REQUESTED** [review]
A few notes on the code itself.

File: src/main/java/fr/moribus/imageonmap/commands/maptool/RenameCommand.java (hunk 3 above)
Comment (Vlammar) [reply src/main/java/fr/moribus/imageonmap/commands/maptool/RenameCommand.java:48]: Added

File: src/main/java/fr/moribus/imageonmap/ImageOnMap.java (hunk 1 above)
Comment (Vlammar) [reply src/main/java/fr/moribus/imageonmap/ImageOnMap.java]: Fixed, i have removed * import

**Comment by Vlammar** [comment]
@AmauryCarrade I think we can do the merge, for the name with a space and the usage of " " for name with space.

**Pull request merged**